core/DataTable.php

```diff
@@ -573,7 +573,13 @@ public function addDataTable(DataTable $tableToSum)
             $this->aggregateRowFromSimpleTable($row);
         } else {
             $columnAggregationOps = $this->getMetadata(self::COLUMN_AGGREGATION_OPS_METADATA_NAME);
-            foreach ($tableToSum->getRows() as $row) {
+            foreach ($tableToSum->getRowsWithoutSummaryRow() as $row) {
+                $this->aggregateRowWithLabel($row, $columnAggregationOps);
+            }
+            // we do not use getRows() as this method might get called 100k times when aggregating many datatables and
+            // this takes a lot of time.
+            $row = $tableToSum->getRowFromId(DataTable::ID_SUMMARY_ROW);
+            if ($row) {
```

Comment:
Could point, I think that would simplify many things and even improve performance sometimes. I might have a look at this end of week or next week. Not sure if I will find time though.